ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/worldstate/Pruner.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.worldstate;
+
+import tech.pegasys.pantheon.ethereum.ProtocolContext;
+import tech.pegasys.pantheon.ethereum.chain.BlockAddedEvent;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.core.Hash;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Pruner {
+  private static final Logger LOG = LogManager.getLogger();
+  private final MarkSweepPruner pruningStrategy;
+  private final ProtocolContext<?> protocolContext;
```

Comment:
@ajsutton can this take a blockchain as a dependency instead? It doesn't seem like the protocolContext is used for anything other than getting the chain

Reply:
Yeah it should only need the blockchain so if it makes it easier to just pass that in go for it.  We have a tendency in a lot of places to pass `ProtocolContext` around as a unit but definitely not a hard and fast rule.  I'm guessing in an earlier version it needed the world state as well, but the `MarkSweepPruner` encapsulates that now.